src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource/DiagnosticSourceEventSourceListener.cs

```diff
@@ -0,0 +1,208 @@
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private static bool IsActivitySourceEntry(string filterAndPayloadSpec, int startIdx, int endIdx) =>
+        filterAndPayloadSpec.AsSpan(startIdx, endIdx - startIdx).StartsWith(c_ActivitySourcePrefix.AsSpan(), StringComparison.Ordinal);
+
+    private static DiagnosticSourceEventSourceFilterAndTransform? CreateNewActivitySourceTransform(string filterAndPayloadSpec, int startIdx, int endIdx, DiagnosticSourceEventSourceFilterAndTransform? next)
+    {
+        Debug.Assert(endIdx - startIdx >= 4);
+        Debug.Assert(IsActivitySourceEntry(filterAndPayloadSpec, startIdx, endIdx));
+
+        ReadOnlySpan<char> eventName;
+        ReadOnlySpan<char> activitySourceName;
+
+        DiagnosticSourceEventSourceFilterAndTransform.ActivityEvents supportedEvent = DiagnosticSourceEventSourceFilterAndTransform.ActivityEvents.All; // Default events
+        DiagnosticSourceEventSourceSamplerBuilder.SampleActivityFunc sampleFunc = static (bool hasActivityContext, ref ActivityCreationOptions<ActivityContext> options)
+            => ActivitySamplingResult.AllDataAndRecorded; // Default sampler
+
+        int colonIdx = filterAndPayloadSpec.IndexOf(':', startIdx + c_ActivitySourcePrefix.Length, endIdx - startIdx - c_ActivitySourcePrefix.Length);
+
+        ReadOnlySpan<char> entry = filterAndPayloadSpec.AsSpan(
+            startIdx + c_ActivitySourcePrefix.Length,
+            (colonIdx >= 0 ? colonIdx : endIdx) - startIdx - c_ActivitySourcePrefix.Length)
+            .Trim();
+
+        int eventNameIndex = entry.IndexOf('/');
+        if (eventNameIndex >= 0)
+        {
+            activitySourceName = entry.Slice(0, eventNameIndex).Trim();
+
+            ReadOnlySpan<char> suffixPart = entry.Slice(eventNameIndex + 1).Trim();
+            int samplingResultIndex = suffixPart.IndexOf('-');
+            if (samplingResultIndex >= 0)
+            {
+                // We have the format "[AS]SourceName/[EventName]-[SamplingResult]
+                eventName = suffixPart.Slice(0, samplingResultIndex).Trim();
+                suffixPart = suffixPart.Slice(samplingResultIndex + 1).Trim();
+
+                if (suffixPart.Length > 0)
+                {
+                    if (suffixPart.Equals("Propagate".AsSpan(), StringComparison.OrdinalIgnoreCase))
+                    {
+                        sampleFunc = static (bool hasActivityContext, ref ActivityCreationOptions<ActivityContext> options) => ActivitySamplingResult.PropagationData;
+                    }
+                    else if (suffixPart.Equals("Record".AsSpan(), StringComparison.OrdinalIgnoreCase))
+                    {
+                        sampleFunc = static (bool hasActivityContext, ref ActivityCreationOptions<ActivityContext> options) => ActivitySamplingResult.AllData;
+                    }
+                    else if (suffixPart.StartsWith(c_ParentRatioSamplerPrefix.AsSpan(), StringComparison.OrdinalIgnoreCase))
```

Comment:
I'm assuming this is the only change in the parsing code?

Reply:
This ended up moving into `DsesFilterAndTransform` but yes, only adding the ability to parse out `ParentRatioSampler` into `DsesFilterAndTransform.SampleFunc`.